File: CHEFSTLT.cpp
#include <iostream>
#include <cstring>
using namespace std;
int main(){
	int t;
	cin>>t;
	char a[100];
	char b[100];
	while(t--){
		cin>>a;
		cin>>b;
		//cout<<a<<b<<endl;
		int min=0;
		int max=0;
		for(int i=0;i<strlen(a);i++){
			if(a[i]!='?' && b[i]!='?' && a[i]!=b[i]){
				min++;
			}
			if(a[i]=='?' || b[i]=='?'){
				max++;
			}
		}
		cout<<min<<" "<<min+max<<endl;
	}
}

File: CHEFSTLT.pl
$t=<>;
while ( $t > 0){

	@str1=split("",<STDIN>);
	@str2=split("",<STDIN>);
	$i=0;
	$min=0;
	$max=0;
	for my $i(0 .. ($#str1/2)+1){
		print "@str1[$i] @str2[$i]\n";
		if(@str1[$i] != '?' and @str2[$i]!='?' and @str1[$i]!=@str2[$i]){
			$min=$min+1;
		}
		if(@str1[$i]=='?' or @str2[$i]=='?'){
			$max=$max+1;
		}
	}
	print "min=$min max=$max\n";
	#print "@str1\n";
	#print "@str2\n";
	$t=$t-1;
}